addOns/httpsInfo/src/main/java/org/zaproxy/zap/extension/httpsinfo/ExtensionHttpsInfo.java

```diff
@@ -119,13 +134,18 @@ public void setVisible(Component component, boolean visible) {
     }
 
     protected AbstractPanel getHttpsInfoPanel() {
-        if (httpsInfoPanel == null) {
+        if (httpsInfoPanel == null && neededUsage) {
```

Comment:
It's simpler if the same panel is returned for both cases, we can make use of the `CardLayout` to swap between the usage and the results panels. In which case we would no longer need `UsagePanel`, it would be simply a `JPanel`.
The usage should be shown after a new session.

Reply:
Okay could you elaborate more about the "swap between usage and results panel"? Do you want to have a tab for usage like other info tabs?

Reply:
No, the usage would only be shown when there are no results. The `CardLayout` allows a container to have several panels with only one being shown at a single time, that means you can easily swap between `httpsInfoTabsPanel` and `usagePanel` when needed (by using, e.g. [`CardLayout#next(Container)`](https://docs.oracle.com/javase/8/docs/api/java/awt/CardLayout.html#next-java.awt.Container-)). rather than manually remove the current panel and add the other one.

Reply:
![Screenshot (15)](https://user-images.githubusercontent.com/40140408/97535936-6b64bf00-19e2-11eb-8b7b-65aaa235ae7c.png)

Initially when I start the httpsInfo extension it looks like this.

Then when a user starts a scan then it is replaced by the result tab like below.

![Screenshot (16)](https://user-images.githubusercontent.com/40140408/97536052-951de600-19e2-11eb-9374-0e92eff49eba.png)

Isn't this what are u trying to say?

Reply:
Yes, but the `CardLayout` comment is not about the behaviour itself but how it is implemented, making use of the `CardLayout` simplifies the swaps. (Note that we should also show the usage panel when the session is changed, as is it would mean removing and adding the panels again, instead of just calling `next`.)

Reply:
Okay so do you want me to remove the getUsagePanel() and just add a text area in line 142?

Reply:
The getter can be kept (and changed to use a plain `JPanel`). Yes, both panels should be added to the `httpsInfoPanel` at the same time (with usage added later), then show the next one (e.g. `((CardLayout) getHttpsInfoPanel().getLayout()).next(getHttpsInfoPanel())`). In `addTab` and `sessionChanged` show/hide the usage panel accordingly.